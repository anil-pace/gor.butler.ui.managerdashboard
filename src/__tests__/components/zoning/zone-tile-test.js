import React from 'react';
import Zone from '../../../components/systemOverview/zoneTile.js';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

const props = {
      name:"Zone 1",
      id:"1",
      emergencyStatus:true,
      emergencyType:"hard",
      statusText:"STOPPED",
      statusClass:"zoneClearInit",
      status:"zone_clear",
      onZoneClick:function(){

      }
    }

test('Result component should display data correctly', () => {

  const component = renderer.create(
    <Zone 
                    name={props.name}
                    id={props.id}
                    emergencyStatus={props.emergencyStatus}
                    emergencyType={props.emergencyType}
                    statusText={props.statusText}
                    status={props.status}
                    statusClass={props.statusClass}
                    onZoneClick={props.onZoneClick}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
 
});

describe('test zone tile component', () => {
it('renders zone name', () => {
  const wrapper = shallow(<Zone 
                    name={props.name}
                    id={props.id}
                    emergencyStatus={props.emergencyStatus}
                    emergencyType={props.emergencyType}
                    statusText={props.statusText}
                    status={props.status}
                    statusClass={props.statusClass}
                    onZoneClick={props.onZoneClick}/>);
  const zoneName = <p className="zone-name">Zone 1</p>;
  const statusName = <p className="operating-status">STOPPED</p>
  // expect(wrapper.contains(welcome)).to.equal(true);
  expect(wrapper.contains(zoneName)).toEqual(true);
  expect(wrapper.contains(statusName)).toEqual(true);
});
})