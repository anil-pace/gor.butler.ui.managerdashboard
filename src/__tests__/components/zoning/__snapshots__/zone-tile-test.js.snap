exports[`test Result component should display data correctly 1`] = `
<div
  className="gor-zone-tile zoneClearInit hard"
  onClick={[Function]}>
  <div
    className="left-content">
    <div
      className="gor-zone-status">
      <span />
    </div>
  </div>
  <div
    className="right-content">
    <p
      className="zone-name">
      Zone 1
    </p>
    <p
      className="operating-status">
      STOPPED
    </p>
    <p />
  </div>
</div>
`;
